core/node/api_server/src/web3/mod.rs

```diff
@@ -285,6 +297,11 @@ impl ApiBuilder {
         self
     }
 
+    pub fn with_main_node_client(mut self, main_node_client: Box<DynClient<L2>>) -> Self {
```

Comment:
Bikeshedding: IMO, it's not clear that setting the client determines whether the bridge addresses task will be enabled. IMO, it'd make sense to accept `BridgeAddressesHandle` directly; this would be more in line with more modular DI we use almost (?) everywhere except for the API server because of legacy reasons.

Reply:
Yep, LGTM. Maybe makes sense to ask @popzxc's opinion as well.